### DISCUSSION TITLE
it is possible to use Ethernet, SDCard SDIO and Trinamic at the same time

### DISCUSSION THREAD

**Discussion by semos2k** [grblHAL/STM32F4xx#162] [General]
Hi, I am using a black STM32F407VET6 board and it has 3 SPI interfaces, it is possible to use Ethernet, SDCard SDIO and Trinamic at the same time ?

thks.

**Comment by terjeio** [comment]
I think so, but I have not tested it, The Trinamic interface is for the most part quiet so should not interfere with anything.
SDIO and Ethernet seems to work nice together - I have just run some lomg term bench tests without issues. And ftp transfers are pretty fast at around 500KB/sec - this with updated driver code that I will soon commit. I have even been running the WebUI - but I am not sure if I can manage to make work completely - the RAM is a bit low at 128 KB blocking fetch of the grblHAL settings...

**Reply by andrewmarles** [reply]
I tried to run webUI as well but it crashed when trying to load the settings.  I don't know if there is some way to split up the output into smaller bites?

**Reply by terjeio** [reply]
Not without changes in the WebUI client. Or could the payload be delivered in chunks? I do not know.
Anyway - not a small change. So perhaps trying to free enough memory should be tried initially?

**Comment by semos2k** [comment]
Its works perfectly, I have tested w5500, along with an sdio sd card and mks tmc2160.

thank you so much
